src/packages/core/src/server.ts

```diff
@@ -99,45 +124,53 @@ export default class Server {
       return callbackIsFunction
         ? process.nextTick(callback!, err)
         : Promise.reject(err);
-    } else if (status & Status.open) {
-      // if open or opening
-      const err = new Error(`Server is already open on port: ${port}.`);
+    } else if ((status & Status.openingOrOpen) !== 0) {
+      // if opening or open
+      const err = new Error(`Server is already open, or is opening, on port: ${port}.`);
       return callbackIsFunction
         ? process.nextTick(callback!, err)
         : Promise.reject(err);
     }
 
     this.#status = Status.opening;
 
-    const promise = new Promise(
-      (resolve: (listenSocket: false | uWS.us_listen_socket) => void) => {
-        // Make sure we have *exclusive* use of this port.
-        // https://github.com/uNetworking/uSockets/commit/04295b9730a4d413895fa3b151a7337797dcb91f#diff-79a34a07b0945668e00f805838601c11R51
-        const LIBUS_LISTEN_EXCLUSIVE_PORT = 1;
-        hostname
-          ? this.#app.listen(
-              hostname,
-              port,
-              LIBUS_LISTEN_EXCLUSIVE_PORT,
-              resolve
-            )
-          : this.#app.listen(port as any, LIBUS_LISTEN_EXCLUSIVE_PORT, resolve);
-      }
-    ).then(listenSocket => {
-      if (listenSocket) {
-        this.#status = Status.open;
-        this.#listenSocket = listenSocket;
-        if (callbackIsFunction) callback!(null);
-      } else {
-        this.#status = Status.closed;
-        const err = new Error(
-          `listen EADDRINUSE: address already in use ${
-            hostname || DEFAULT_HOST
-          }:${port}.`
-        );
-        if (callbackIsFunction) callback!(err);
-        else throw err;
-      }
+    const initializePromise = this.initialize();
+
+    const promise = initializePromise.then(() => {
```

Comment:
I think we should still parallelize chain initialization and server start up here. I think it's still possible? So, instead of `initializePromise.then(...)`, perhaps we can `const promise = Promise.allSettled([initializePromise, new Promise(...)).catch(...);` (note: for this approach you'll probably need to use the `promise.allsettled` polyfill because of our Node 10 support)